source/extensions/filters/http/ext_proc/matching_utils.cc

```diff
@@ -0,0 +1,84 @@
+#include "source/extensions/filters/http/ext_proc/matching_utils.h"
+
+#include <memory>
+
+#if defined(USE_CEL_PARSER)
+#include "parser/parser.h"
+#endif
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+absl::flat_hash_map<std::string, Filters::Common::Expr::ExpressionPtr>
+ExpressionManager::initExpressions(const Protobuf::RepeatedPtrField<std::string>& matchers) {
+  absl::flat_hash_map<std::string, Filters::Common::Expr::ExpressionPtr> expressions;
+#if defined(USE_CEL_PARSER)
+  for (const auto& matcher : matchers) {
+    auto parse_status = google::api::expr::parser::Parse(matcher);
+    if (!parse_status.ok()) {
+      throw EnvoyException("Unable to parse descriptor expression: " +
+                           parse_status.status().ToString());
+    }
+
+    auto& iter = expr_list_.emplace_back(parse_status.value());
```

Comment:
There is inconsistency here: `parsed expression` will be stored in `expr_list_` for both request and response together, while the `compiled_expression` will be stored for request and response, respectively.  For example, this inconsistency could lead different behaviors in skip insertion as I comment above

How about creating new struct(like below) and use that as your hash value `absl::flat_hash_map<std::string, CelExpression>`
```
struct CelExpression {
    ParsedExpr parsed_expr_;
    ExpressionPtr compiled_expr_;
  };
 ```